packages/svelte/src/compiler/preprocess/replace_in_code.js

```diff
@@ -20,7 +20,7 @@ export function slice_source(code_slice, offset, { file_basename, filename, get_
  * @param {(...match: any[]) => Promise<MappedCode>} get_replacement
  * @param {string} source
  */
-function calculate_replacements(re, get_replacement, source) {
+async function calculate_replacements(re, get_replacement, source) {
```

Comment:
I don't understand. What is this for?

Reply:
Because it returns a promise

Reply:
so?

Reply:
(to be clear: I understand that eslint fails because of `@typescript-eslint/promise-function-async`, but that rule seems particularly pointless even by the standards of the genre)

Reply:
(in fact i'm pretty sure that making a function `async` unnecessarily creates runtime overhead. i just have no idea why we would do this to ourselves)

Reply:
Marking a function as async helps consumers know that they should `await` it. What's the reason we would omit it or why do you say it's unnecessary?

Reply:
If the types say that it returns a promise, then consumers know they need to await it anyway.

Reply:
I still don't understand the benefit of leaving it off. Should we remove `async` from all functions in the codebase?

Reply:
All of the ones that don't have an `await` in them, probably, yeah.

Reply:
> I still don't understand the benefit of leaving it off

As mentioned, it creates runtime overhead. AFAIK this will result in four promises being created rather than two:

```js
async function get(thing) {
  return fetch(`${base}/${thing}`).then(async (r) => r.json());
}
```

Reply:
> ...to tell the plugin that Promise.resolve() — of all things! — doesn't need a .catch handler tells you all you need to know about how much value it adds.

I believe that the issue isn't `Promise.resolve()`, but that an exception could be thrown in `.then()` and then it would be unhandled

Reply:
Either way it's an annoying rule that we really don't need. I got rid of the ones that don't provide any value (or provide negative value, as in the case of `promise-function-async`)